import { render } from '@testing-library/react';
import Error from '.';
import userEvent from '@testing-library/user-event';

/*
  beforeEach(() => {
    console.log('yeni test baÅŸalamadan Ã¶nce ðŸ˜€ðŸ˜€');
  });

  afterEach(() => {
    console.log('test bittikten sonra ðŸ˜‚ðŸ˜‚');
  });

  beforeAll(() => {
    console.log('bÃ¼tÃ¼n testlerden Ã¶nce ðŸ˜€ðŸ˜€');
  });

  afterAll(() => {
    console.log('bÃ¼tÃ¼n testlerden sonra ðŸ˜€ðŸ˜€');
  });
*/

describe('error bielÅŸeni testleri', () => {
  // gerekli kurumlar
  const user = userEvent.setup();
  const mockFn = jest.fn();
  let comp;

  //  her testin Ã¶ncesinde error bileÅŸeni render edilir
  beforeEach(() => {
    comp = render(
      <Error message={'Failed with status code of 404'} retry={mockFn} />
    );
  });

  // TEST - 1
  it('doÄŸru hata mesajÄ±nÄ± gÃ¶sterir', () => {
    comp.getByText(/failed with/i);
  });

  // TEST - 2
  it('tekrar dene butonu Ã§alÄ±ÅŸÄ±r', async () => {
    const button = comp.getByRole('button');

    await user.click(button);

    expect(mockFn).toHaveBeenCalled();
  });
});
